File: backend/Sources/backend/DTO/StationDTO.swift
//
//  StationDTP.swift
//  backend
//
//  Created by Christian Knapp on 28.09.25.
//

import Vapor

struct StationDTO: Content {
    let evaNumber: Int
    let name: String
    let position: PositionDTO
    let ril100: [String]
    let transports: [TransportType]
    
    func toModel() -> Station {
        Station(
            id: self.evaNumber,
            name: self.name,
            weight: 0,
            latitude: self.position.latitude,
            longitude: self.position.longitude,
            queryingEnabled: false,
            lastQueried: nil,
            isLocked: false
        )
    }
}

File: backend/Navigator.Api/DTOs/Station/StationDTO.cs
using Navigator.Data.Enums;
using System.Text.Json.Serialization;

namespace Navigator.Api.DTOs.Station;

public class StationDTO
{
    [JsonPropertyName("evaNumber")]
    public required int EvaNumber { get; set; }

    [JsonPropertyName("name")]
    public required string Name { get; set; }

    [JsonPropertyName("position")]
    public required PositionDTO Position { get; set; }

    [JsonPropertyName("transports")]
    public required TransportType[] Transports { get; set; }

    [JsonPropertyName("ril100")]
    public string[]? Ril100 { get; set; }
}
